connection.go

```diff
@@ -830,19 +835,27 @@ func (conn *Connection) reader(r io.Reader, c Conn) {
 				conn.opts.Logger.Report(LogWatchEventReadFailed, conn, err)
 			}
 			continue
-		} else if resp.Code == PushCode {
-			if fut = conn.peekFuture(resp.RequestId); fut != nil {
-				fut.AppendPush(resp)
+		} else if code == iproto.IPROTO_CHUNK {
+			if fut = conn.peekFuture(header.RequestId); fut != nil {
+				if err := fut.AppendPush(header, &buf); err == nil {
+					err = ClientError{
+						ErrProtocolError,
+						fmt.Sprintf("failed to append push response: %s", err),
+					}
+					conn.opts.Logger.Report(LogAppendPushFailed, conn, err)
+				}
 			}
 		} else {
-			if fut = conn.fetchFuture(resp.RequestId); fut != nil {
-				fut.SetResponse(resp)
+			if fut = conn.fetchFuture(header.RequestId); fut != nil {
+				if err := fut.SetResponse(header, &buf); err != nil {
+					fut.SetError(fmt.Errorf("failed to set response: %w", err))
```

Comment:
Could you please add a test for the case? (with a custom request type)

Reply:
New test was added: `TestConnection_SetResponse_failed` in the `tarantool_test.go`

Reply:
https://github.com/tarantool/go-tarantool/blob/aed12331cccd611cf87e78074320e95f6278d6b6/tarantool_test.go#L1814-L1824

Reply:
Cool, thank you!